src/Uno.UI/UI/Xaml/FrameworkElementExtensions.cs

```diff
@@ -324,6 +324,48 @@ internal static bool TrySetPadding(this IFrameworkElement frameworkElement, Thic
 			return false;
 		}
 
+		internal static bool TryGetActualBorderThickness(this IFrameworkElement frameworkElement, out Thickness borderThickness)
+		{
+			if (__LinkerHints.Is_Windows_UI_Xaml_Controls_Grid_Available && frameworkElement is Grid g)
+			{
+				borderThickness = g.BorderThickness;
+				return true;
+			}
+
+			if (__LinkerHints.Is_Windows_UI_Xaml_Controls_StackPanel_Available && frameworkElement is StackPanel sp)
+			{
+				borderThickness = sp.BorderThickness;
+				return true;
+			}
+
+			if (__LinkerHints.Is_Windows_UI_Xaml_Controls_RelativePanel_Available && frameworkElement is RelativePanel rp)
+			{
+				borderThickness = rp.BorderThickness;
+				return true;
+			}
+
+			if (__LinkerHints.Is_Microsoft_UI_Xaml_Controls_LayoutPanel_Available && frameworkElement is Microsoft.UI.Xaml.Controls.LayoutPanel lp)
+			{
+				borderThickness = lp.BorderThickness;
+				return true;
+			}
+
+			if (__LinkerHints.Is_Windows_UI_Xaml_Controls_ContentPresenter_Available && frameworkElement is ContentPresenter cp)
+			{
+				borderThickness = cp.BorderThickness;
+				return true;
+			}
+
+			if (__LinkerHints.Is_Windows_UI_Xaml_Controls_Border_Available && frameworkElement is Border b)
+			{
+				borderThickness = b.BorderThickness;
+				return true;
+			}
+
+			borderThickness = default;
+			return false;
+		}
```

Comment:
Is this missing `Control.BorderThickness`?

https://github.com/unoplatform/uno/blob/0190ea92ffc138e1cc3259e59fdf2074cbc16465/src/Uno.UI/UI/Xaml/Controls/Control/Control.cs#L630

Reply:
In this case it is intentional - only for `Panels`, `ContentPresenter` and `Border` the `BorderThickness` is actually "meaningful". `Control.BorderThickness` does nothing unless it is applied on something within its control template.